src/lib/storage/proposal/partition_creator.rb

```diff
@@ -281,14 +281,27 @@ def create_partition(vol, partition_id, free_slot)
           log.info("Creating partition for #{vol.mount_point} with #{vol.size}")
           disk = ::Storage::Disk.find(devicegraph, free_slot.disk_name)
           ptable = disk.partition_table
-          if logical_partition_preferred?(ptable)
-            create_extended_partition(disk, free_slot.slot.region) unless ptable.has_extended
+
+          logical = false
+          if space_inside_extended?(free_slot)
+            logical = true
+          elsif logical_partition_preferred?(ptable)
+            # It's not possible to have more than one extended partition,
+            # even if logical partitions are preferred
+            if !ptable.has_extended
+              create_extended_partition(disk, free_slot.slot.region)
+              logical = true
+            end
+          end
```

Comment:
@wfeldt Actually, the current code simply can't handle more than one space. See PR in https://trello.com/c/Yvl7G5tB/ for initial experiments about it... and that case is indeed something worth adding to the list of scenarios in that PBI,